nemo/collections/tts/modules/common.py

```diff
@@ -232,24 +232,35 @@ def masked_conv_to_sequence(self, context: Tensor, lens: Tensor, enforce_sorted:
         )
         return seq
 
-    def forward(self, context: Tensor, lens: Optional[Tensor] = None) -> Tensor:
-        if lens is None:
-            for conv in self.convolutions:
-                context = self.dropout(F.relu(conv(context)))
-            context = context.transpose(1, 2)
-            context, _ = self.bilstm(context)
-        else:
-            # borisf : does not match ADLR (values, lengths)
-            # seq = self.masked_conv_to_sequence(context, lens, enforce_sorted=False)
-            # borisf : does match ADLR
-            seq = self.conv_to_sequence(context, lens, enforce_sorted=False)
-            context, _ = self.lstm_sequence(seq)
+    def forward(self, context: Tensor, lens: Tensor) -> Tensor:
```

Comment:
Let's is optional apparently in types, maybe TTS folks need careful review of this pr - @redoctopus @XuesongYang

Reply:
What about making it non optional in forward_for_export()

Reply:
Why? This module is only used with lengths. I can restore Optional though, just in case.
Those changes were made in collaboration with @MikyasDesta and Kevin Shih  - RADTTS authors from ADLR, and tested on training and export.

Reply:
If the `lens is None` handling is being removed, I think we should get rid of `Optional` just to minimize confusion. As long as we're certain that `lens` will always be passed in this is fine.

Reply:
I just added it back just in case.

Reply:
> Why? This module is only used with lengths. I can restore Optional though, just in case. Those changes were made in collaboration with @MikyasDesta and Kevin Shih - RADTTS authors from ADLR, and tested on training and export.

a side note: there are not any unit tests and CI tests for RADTTS training and export in NeMo repo, which may have risks. I suggested adding tests when porting it from ADLR (https://github.com/NVIDIA/NeMo/pull/4538#pullrequestreview-1061119710), but they are still missing.